src/proc_diskstats.c

```diff
@@ -523,6 +678,61 @@ int do_proc_diskstats(int update_every, unsigned long long dt) {
 		}
 
 		// --------------------------------------------------------------------
+
+		if(ddo_space) {
+			if(mount_point) {
+				if (statvfs(family, &buff_statvfs) < 0) {
+					error("Faild checking disk space usage of %s", family);
+				} else {
+					space_avail = buff_statvfs.f_bavail * buff_statvfs.f_bsize;
+					space_avail_root = (buff_statvfs.f_bfree - buff_statvfs.f_bavail) * buff_statvfs.f_bsize;
+					space_used = (buff_statvfs.f_blocks - buff_statvfs.f_bfree) * buff_statvfs.f_bsize;
+				}
+
+				st = rrdset_find_bytype("disk_space", disk);
+				if(!st) {
+					unsigned long long space_total = buff_statvfs.f_blocks * buff_statvfs.f_bsize;
+					unsigned long long unit_div;
+					unsigned long long k = 1000;
+					char* unit;
+					if (space_total < (1000)) {
+						unit_div = 1;
+						unit = "B";
+					} else if (space_total < (k*k)) {
+						unit_div = k;
+						unit = "KB";
+					} else if (space_total < (k*k*k)) {
+						unit_div = k*k;
+						unit = "MB";
+					} else if (space_total < (k*k*k*k)) {
+						unit_div = k*k*k;
+						unit = "GB";
+					} else if (space_total < (k*k*k*k*k)) {
+						unit_div = k*k*k*k;
+						unit = "TB";
+					}
```

Comment:
I do not calculate the unit with the amout of space free. I use total disk space.
My idea was that total disk space only changes when changing the disk. This is wrong. The size of containers and partitions can varry quite more.

Until having something that can handle to display matching units on the client side which should I use?
Megabyte or Gigabyte? I would prefer Gigabyte.